core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -460,11 +462,16 @@ public BigInteger getDifficultyTargetAsInteger() {
         return difficultyTarget.asInteger();
     }
 
-    /** Returns true if the hash of the block is OK (lower than difficulty target). */
-    protected boolean checkProofOfWork(boolean throwException) throws VerificationException {
+    /**
+     * Checks the proof of work of this block. If this method returns without throwing, the hash of this block is OK
+     * (lower than difficulty target).
+     *
+     * @throws VerificationException if there is not enough proof of work
+     */
+    private void checkProofOfWork() throws VerificationException {
```

Comment:
I would consider changing this to always return a `boolean` and have higher-level methods decide to throw an exception.

Reply:
Actually, this *is* the higher level method that decides to throw. The method with a boolean is now in Difficulty.